General fixes to install script

directhex opened the pull request:
Stop installing way more stuff than we need to

radical commented:
We seem to be missing some files in the `Output` directory like `Microsoft.{CSharp,VisualBasic}.Core.targets`, `csc.exe`(and friends) etc. Also, `*UnitTests*` assemblies are present even in this, so, those should still be removed.

directhex commented:
Do we _want_ csc.exe? Are we expecting that building msbuild from source bundles Roslyn from NuGet?

Commit:
Don't assume OSX_Deployment on installation

This could be Unix_Deployment on Linux

Commit:
Support DESTDIR on installation

Most build systems allow a DESTDIR environment variable, for staging an installation from one location to another. For example, if Mono is in `/usr/lib/mono` but you want to copy the installation files to a temporary staging directory of `/tmp/foo`, `DESTDIR=/tmp/foo ./install-mono-prefix.sh /usr` will install the files to `/tmp/foo` but symlinks and modifications to `bin/msbuild` will look for Mono etc in `/usr`, not `/tmp/foo`

Commit:
Use wildcard for cp calls on installation

`cp foo/ /bar/` will install `foo` itself into `/bar` in some cases, rather than the *contents* of `foo`. Explicitly using a wildcard means it always does the right thing.

radical commented on install-mono-prefix.sh:
```diff
@@ -6,26 +6,27 @@ fi
 sed -e 's,@bindir@,'$MONO_PREFIX'/bin,' -e 's,@mono_instdir@,'$MONO_PREFIX/lib/mono',' msbuild-mono-deploy.in > msbuild-mono-deploy.tmp
 chmod +x msbuild-mono-deploy.tmp
-cp msbuild-mono-deploy.tmp $MONO_PREFIX/bin/msbuild
+cp msbuild-mono-deploy.tmp ${DESTDIR}${MONO_PREFIX}/bin/msbuild
```
And add a `rm msbuild-mono-deploy.tmp`. I forgot to add that earlier!

radical commented:
Other than the minor comment, LGTM! 👍

Commit:
Delete temporary file
